### PR TITLE
rescan_datastores: refactoring

### PR DESCRIPTION
**Pull request by rkhozinov**

### CODE REVIEW

**Review by ekhomyakova — CHANGES_REQUESTED** [review]
I don't like it

File: servers/partner-ci/builders/rescan_ds.py [hunk 1]
```diff
@@ -47,360 +41,83 @@ class Victl(object):
...
-
-    return 0
-
-
-def check_portgroup(args, inst):
-    """Return 0 if portgroup is configured on cluster."""
-    if inst.check_portgroup_configured(args.datacenter, args.cluster,
-                                       args.portgroup):
+        self.dc_name = dc_name
+        self.datacenter = dc_name
```
Comment (ekhomyakova) [inline servers/partner-ci/builders/rescan_ds.py]: What is the difference between `self.dc_name` and `self.datacenter`?

**Review by ekhomyakova — CHANGES_REQUESTED** [review]

File: servers/partner-ci/builders/rescan_ds.py [hunk 2]
```diff
@@ -47,360 +41,83 @@ class Victl(object):
...
+            host_view = self.content.viewManager.CreateContainerView(self.datacenter,
+                                                                     [vim.HostSystem],
+                                                                     True)
+            self._hosts = [host for host in host_view.view ]
+            host_view.Destroy()
+        return self._hosts
+
+    def rescan_datastores(self):
+        """Rescan all datastores."""
+        [host.configManager.storageSystem.RescanAllHba() for host in self.hosts]
```
Comment (ekhomyakova) [inline servers/partner-ci/builders/rescan_ds.py]: Don't create new object. There is nothing bad in usual loop.

File: servers/partner-ci/builders/rescan_ds.py [hunk 3]
```diff
@@ -47,360 +41,83 @@ class Victl(object):
...
+                                    "with '{0}' name".format(value))
+
+    @property
+    def hosts(self):
+
+        if not self._hosts:
+            host_view = self.content.viewManager.CreateContainerView(self.datacenter,
+                                                                     [vim.HostSystem],
+                                                                     True)
+            self._hosts = [host for host in host_view.view ]
```
Comment (ekhomyakova) [inline servers/partner-ci/builders/rescan_ds.py]: Redundant space

File: servers/partner-ci/builders/rescan_ds.py [hunk 4]
```diff
@@ -47,360 +41,83 @@ class Victl(object):
...
+        self._datacenter = [dc for dc
+                            in self.content.rootFolder.childEntity
+                            if dc.name == value][0]
+        if not self._datacenter:
+            raise NotFoundException("There's no datacenter "
+                                    "with '{0}' name".format(value))
+
+    @property
+    def hosts(self):
+
```
Comment (ekhomyakova) [inline servers/partner-ci/builders/rescan_ds.py]: Remove the line

**Review by ekhomyakova — CHANGES_REQUESTED** [review]

File: servers/partner-ci/builders/rescan_ds.py [hunk 5]
```diff
@@ -47,360 +41,83 @@ class Victl(object):
...
+        if not self._hosts:
+            host_view = self.content.viewManager.CreateContainerView(self.datacenter,
```
Comment (ekhomyakova) [inline servers/partner-ci/builders/rescan_ds.py]: Move `self.datacenter,` to the new line.

File: servers/partner-ci/builders/rescan_ds.py [hunk 6]
```diff
@@ -47,360 +41,83 @@ class Victl(object):
...
+        """Print list of datastores."""
+        for host in self.hosts:
+            log.info("  On esxi '{0}' datastores:".format(host.name))
+            for ds in host.datastore:
+                log.info("\t'{0}'".format(ds.name))
 
 
 if __name__ == '__main__':
-    args = Params()
-    inst = Victl(args.host, args.user, args.password, args.port)
```
Comment (ekhomyakova) [inline servers/partner-ci/builders/rescan_ds.py]: Remove the line

File: servers/partner-ci/builders/rescan_ds.py [hunk 7]
```diff
@@ -47,360 +41,83 @@ class Victl(object):
...
+    password = environ.get('VCENTER_PASSWORD', 'vmware')
+    dc_name = environ.get('VC_DATACENTER', 'Datacenter')
+    datastore = environ.get('VC_DATASTORE', 'nfs')
+    clusters = environ.get('VCENTER_CLUSTERS', 'Cluster1')
 
-    for cluster in args.clusters.split(','):
-        args.cluster = cluster
-        datastore_list(args, inst)
+    victl = Victl(host=host, dc_name=dc_name, user=user,
+                    password=password, port=port)
```
Comment (ekhomyakova) [inline servers/partner-ci/builders/rescan_ds.py]: Edit formatting

Comment (ekhomyakova) [inline servers/partner-ci/builders/rescan_ds.py]: Add empty line after last one.

**Review by ekhomyakova — CHANGES_REQUESTED** [review]

**Comment by rkhozinov** [comment]
recheck

File: servers/partner-ci/builders/rescan_ds.py (hunk 6 above)
Comment (rkhozinov) [reply servers/partner-ci/builders/rescan_ds.py]: done

File: servers/partner-ci/builders/rescan_ds.py [hunk 8]
```diff
@@ -21,7 +21,8 @@
 import sys
 from os import environ,path
 
-sys.path.append(path.join(sys.path[0], "{0}/bin".format(environ.get('VIRTUAL_ENV'))))
+jenkins_venv = "/home/jenkins/90-venv/bin/activate_this.py"
+execfile(jenkins_venv, dict(__file__=jenkins_venv))
```
Comment (rkhozinov) [inline servers/partner-ci/builders/rescan_ds.py]: awesome

File: .travis.yml [hunk 9]
```diff
@@ -8,18 +8,20 @@ python:
...
+      - debian-sid    # Grab shellcheck from the Debian repo (o_O)
     packages:
       - shellcheck
 
-
 script:
   - bash -c 'shopt -s globstar; shellcheck -e SC1091,SC2086,SC2046,SC2015,SC2155,SC1090,SC2012,SC2035,SC2016 **/*.sh'
   - tox -e 9.x,9.1,9.0
-  - bash -c 'grep -m1 -n -o --color -R -E "[^$]{(\w\w+)}" ./output && exit 1 || echo "there is no undefined variables"'
+  - bash -c 'grep --color -R -E "[^$]{[a-zA-Z0-9_-]+}" ./output && exit 1 || echo "there is no undefined variables"'
```
Comment (rkhozinov) [inline .travis.yml]: not good 
that merge has broken travis

File: .travis.yml [hunk 10]
```diff
@@ -8,18 +8,20 @@ python:
 install:
   - sudo pip install --upgrade pip
   - sudo pip install tox
+  - sudo pip install flake8
 
 addons:
   apt:
     sources:
-      - debian-sid    # Grab shellcheck from the Debian repo
+      - debian-sid    # Grab shellcheck from the Debian repo (o_O)
```
Comment (ekhomyakova) [inline .travis.yml]: Remove " (o_O)"

File: servers/partner-ci/builders/rescan_ds.py [hunk 11]
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env python
+
```
Comment (ekhomyakova) [inline servers/partner-ci/builders/rescan_ds.py]: Is it needed?

File: servers/partner-ci/builders/rescan_ds.py [hunk 12]
```diff
@@ -14,26 +15,24 @@
...
 import atexit
 import logging as log
 import ssl
 import sys
-
 from os import environ
+from os import path
 
-import paramiko
+jenkins_venv = "/home/jenkins/90-venv/bin/activate_this.py"
```
Comment (ekhomyakova) [inline servers/partner-ci/builders/rescan_ds.py]: Don't hardcode path

File: servers/partner-ci/builders/rescan_ds.py [hunk 13]
```diff
@@ -47,360 +46,74 @@ class Victl(object):
...
+        return self._datacenter
 
-        # Build the url to put the file - https://hostname:port/resource?params
-        resource = '/folder/test_upload'
-        params = {'dsName': datastore, 'dcPath': datacenter}
-        http_url = 'https://{0}:443{1}'.format(host, resource)
+    @property
+    def hosts(self):
+        if not self._hosts:
+            host_view = self.content.viewManager.CreateContainerView(self.datacenter,
```
Comment (ekhomyakova) [inline servers/partner-ci/builders/rescan_ds.py]: Move `self.datacenter,` to the next line

File: servers/partner-ci/builders/rescan_ds.py (hunk 12 above)
Comment (rkhozinov) [inline servers/partner-ci/builders/rescan_ds.py]: @naumvd95 remove hardcode for virtual_env
use VIRTUAL_ENV variable instead

File: servers/partner-ci/builders/rescan_ds.py [hunk 14]
```diff
@@ -47,360 +46,74 @@ class Victl(object):
...
-
-
-def check_datastore(args, inst):
-    """Return 0 if datastore is configured on cluster."""
-    inst.write_test_datastore(args.datacenter, args.datastore, args.host)
-    if inst.check_storage_configured(args.datacenter, args.cluster,
-                                     args.datastore):
+    def rescan_datastores(self):
+        """Rescan all datastores."""
+        [host.configManager.storageSystem.RescanAllHba() for host in self.hosts]
```
Comment (ekhomyakova) [inline servers/partner-ci/builders/rescan_ds.py]: Make it in a loop. Generators are not suitable in this case

**Review by ekhomyakova — CHANGES_REQUESTED** [review]

File: servers/partner-ci/builders/common.prepare.sh [hunk 15]
```diff
@@ -52,7 +52,7 @@ fi
 export REQUIRED_FREE_SPACE=200
 
 export ENV_NAME="${ENV_PREFIX}.${SNAPSHOTS_ID}"
-export VENV_PATH="${HOME}/${FUEL_RELEASE}-venv"
+[ -z ${VENV_PATH} ] && { export VENV_PATH="${HOME}/${FUEL_RELEASE}-venv"; }
```
Comment (rkhozinov) [inline servers/partner-ci/builders/common.prepare.sh]: `export VENV_PATH=${VENV_PATH:-"$HOME/$FUEL_RELEASE-venv"}`

File: servers/partner-ci/builders/rescan_ds.py (hunk 11 above)
Comment (rkhozinov) [reply servers/partner-ci/builders/rescan_ds.py]: @naumvd95 please remove that

File: servers/partner-ci/builders/common.prepare.sh [hunk 16]
```diff
@@ -251,3 +251,6 @@ fi
 [[ "${DEBUG}" == "true" ]] && virsh list --all
 sudo cp /var/log/libvirt/libvirtd.log ${WORKSPACE}/libvirtd_before_test.log
 sudo chown jenkins:jenkins ${WORKSPACE}/libvirtd_before_test.log
+
```
Comment (rkhozinov) [inline servers/partner-ci/builders/common.prepare.sh]: remove empty lines

File: servers/partner-ci/builders/rescan_ds.py [hunk 17]
```diff
@@ -14,26 +15,23 @@
...
 import atexit
 import logging as log
 import ssl
 import sys
-
 from os import environ
-
-import paramiko
-
+from os import path
```
Comment (rkhozinov) [inline servers/partner-ci/builders/rescan_ds.py]: add an empty line

File: servers/partner-ci/builders/rescan_ds.py (hunk 14 above)
Comment (rkhozinov) [reply servers/partner-ci/builders/rescan_ds.py]: Please link a proof

File: servers/partner-ci/common.yaml [hunk 18]
```diff
@@ -128,6 +128,7 @@
             SNAPSHOT_OUTPUT_FILE=extra_repos.sh;
             WORKSTATION_PASSWORD=VMware01;
             WORKSTATION_USERNAME=vmware;
+            VENV_PATH=/home/jenkins/90-venv;
```
Comment (rkhozinov) [inline servers/partner-ci/common.yaml]: try to use `$HOME` instead

File: servers/partner-ci/builders/rescan_ds.py (hunk 13 above)
Comment (rkhozinov) [reply servers/partner-ci/builders/rescan_ds.py]: flake check has passed with openstack guidelines

File: servers/partner-ci/builders/rescan_ds.py [hunk 19]
```diff
@@ -14,26 +14,25 @@
...
 """
 
-import argparse
 import atexit
 import logging as log
 import ssl
 import sys
-
 from os import environ
+from os import path
```
Comment (ekhomyakova) [inline servers/partner-ci/builders/rescan_ds.py]: Seems you don't use it.

File: servers/partner-ci/builders/rescan_ds.py [hunk 20]
```diff
@@ -47,360 +46,73 @@ class Victl(object):
 
     _service_instance = None
     content = None
+    _datacenter = None
+    _hosts = []
+    _clusters = []
```
Comment (ekhomyakova) [inline servers/partner-ci/builders/rescan_ds.py]: Seems you don't use it

**Review by naumvd95 — APPROVED** [review]

**Review by rkhozinov — COMMENTED** [review]
looks good for me

**Review by ekhomyakova — APPROVED** [review]

**Review by ekhomyakova — APPROVED** [review]

**Pull request merged**